### DISCUSSION TITLE
AttributeError: 'DataFrame' object has no attribute 'sketch'

### DISCUSSION THREAD

**Discussion by aditya-lawankar** [approximatelabs/sketch#28] [Q&A] [answered]
Hi, 
I am trying to make a frontend using streamlit for querying csv data.
Using sketch on a dataframe (eg: df.sketch.ask()) works fine in a .ipynb notebook.
However, doing the same in .py file for making the frontend gives the following error:
`AttributeError: 'DataFrame' object has no attribute 'sketch'`
which I believe is an error raised from pandas.

Relevant code:

```
import requests
import json
import pandas as pd
import sketch
import matplotlib.pyplot as plt
import streamlit as st

''' Code for reading and cleaning csv '''

st.write(df.head(10))  #works fine, shows the output in frontend

query = st.text_input("Enter query here:")
st.write(df.sketch.ask(query))
```

How do I resolve this?

**Comment by lagane-78** [comment]
I am seeing the same issue ? Did anyone provide any answers to this ?

Thanks

**Comment by bluecoconut** [comment, marked as answer]
This is now fixed! 

if you pull the latest, then this streamlit app will work

```
import requests
import json
import pandas as pd
import sketch
import matplotlib.pyplot as plt
import streamlit as st


''' Code for reading and cleaning csv '''

df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})

st.write(df.head(10))  #works fine, shows the output in frontend

query = st.text_input("Enter query here:")
st.write(df.sketch.ask(query, call_display=False))

```
